docs/java/features/odata/overview.md

```diff
@@ -36,3 +36,24 @@ The SDK supports both OData v2 and OData v4 protocols. However, parts of the v4
 :::tip Improved OData type-safe client
 The Cloud SDK for Java has published an improved OData type-safe client implementation (Beta) which boosts performance of your Java apps and lets you profit from faster innovations for both OData protocols. Check out the [our article on type-safe client for details](use-typed-odata-v2-and-v4-client-for-java) for details.
 :::
+
+## Popular use-cases for Type-safe OData client
+
+### You are extending an SAP product or service, building a middle-ware, publishing a cloud App
+
+- **Type-safe consumption of OData services.** By consumption, we mean building and executing API calls against OData service with serialized request data and processing serialized responses. All in a type-safe way.
+- **Converting OData API into a different flavor of REST API.** You can use SAP Cloud SDK for Java to build an App that converts your desired REST format to OData calls and vice versa. Such App may serve as a proxy, integration adaptor, service gateway, etc.
+- **Serving as a layer to fetch data from and push data to OData services.** Instead of building your query tool you can fetch the data with SAP Cloud SDK and pass it on to your UX framework, analytics tool, data converter, etc. Reverse data flow will allow you to push new or modified data to the OData service for processing or persistence.
```

Comment:
```suggestion
- **Serving as a layer to fetch data from and push data to OData services.** Instead of building your query tool you can fetch the data with SAP Cloud SDK and pass it on to your UX framework, analytics tool, data converter, etc. Reverse data flow will allow you to push new or modified data to the OData service for processing or persistence. In that manner, the SAP Cloud SDK can help  you implement adapters for a [hexagonal architecture](https://en.wikipedia.org/wiki/Hexagonal_architecture_(software)).
```